partcad-cli/src/partcad_cli/click/commands/adhoc/convert.py

```
@@ -0,0 +1,107 @@
+        # Validate input and output formats
+        input_type = input or input_file.suffix.lstrip(".").lower()
+        if not CADConverter.is_supported_format(input_type) or not CADConverter.is_supported_format(output_type):
+            raise click.UsageError(
+                f"Conversion from {input_type} to {output_type} is not supported."
+            )
+
+        # Perform the conversion
+        try:
+            CADConverter.convert(str(input_file), str(output_file), input_type, output_type)
+            logging.info(f"Successfully converted {input_file} to {output_file}.")
+        except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)